File: 112/hazumi.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-


def is_hazumi(a):
    if not (a == sorted(a) or a == sorted(a, reverse=True)):
        return True


if __name__ == "__main__":

    x = 0
    i = 0
    while True:
        i += 1
        a = list(str(i))
        if is_hazumi(a):
            x += 1
            if x * 1.0 / i == 0.99:
                print i
                break

"""
$ time python hazumi.py 
1587000

real    0m9.243s
user    0m9.151s
sys     0m0.046s
"""

File: 112/hazumi.rb
#!/usr/bin/env ruby

def hazumi?
  b = @a.sort
  @a != b && @a != b.reverse
end

x = 0
i = 0

loop do
  i += 1
  @a = i.to_s.split('')
  next unless hazumi?
  x += 1
  if (x * 1.0 / i) == 0.99
    puts i
    break
  end
end

=begin
$ ruby -v
ruby 2.3.0p0 (2015-12-25 revision 53290) [x86_64-darwin14]
$ time ruby hazumi.rb
1587000

real    0m21.346s
user    0m20.645s
sys     0m0.549s
=end
